From: Davide Beatrici <git@davidebeatrici.dev>
Subject: [PATCH] compress: support system miniz library through pkgconfig

---
 vlib/compress/compress.c.v | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/vlib/compress/compress.c.v b/vlib/compress/compress.c.v
index bc0c350ba..8a52889e8 100644
--- a/vlib/compress/compress.c.v
+++ b/vlib/compress/compress.c.v
@@ -1,6 +1,11 @@
 module compress
 
-#flag -I @VEXEROOT/thirdparty/zip
+$if $pkgconfig('miniz') {
+	#pkgconfig miniz
+} $else {
+	#flag -I @VEXEROOT/thirdparty/zip
+}
+
 #include "miniz.h"
 
 pub const max_size = u64(1 << 31)
-- 
2.35.1

