diff --git a/vlib/stbi/stbi.c.v b/vlib/stbi/stbi.c.v
index dbadf43fd..181a7e7ed 100644
--- a/vlib/stbi/stbi.c.v
+++ b/vlib/stbi/stbi.c.v
@@ -4,11 +4,16 @@
 
 module stbi
 
-#flag -I @VEXEROOT/thirdparty/stb_image
+$if $pkgconfig('stb') {
+	#pkgconfig stb
+} $else {
+	#flag -I @VEXEROOT/thirdparty/stb_image
+	#flag @VEXEROOT/thirdparty/stb_image/stbi.o
+}
+
 #include "stb_image.h"
 #include "stb_image_write.h"
 #include "stb_v_header.h"
-#flag @VEXEROOT/thirdparty/stb_image/stbi.o
 
 pub struct Image {
 pub mut:
diff --git a/vlib/v/tests/cflags/vmodroot_and_vroot_test.v b/vlib/v/tests/cflags/vmodroot_and_vroot_test.v
index 182568abf..ed6928e9f 100644
--- a/vlib/v/tests/cflags/vmodroot_and_vroot_test.v
+++ b/vlib/v/tests/cflags/vmodroot_and_vroot_test.v
@@ -8,7 +8,12 @@
 fn C.add(int, int) int
 
 // Tests that VROOT works no matter the current folder
-#flag -I @VEXEROOT/thirdparty/stb_image
+$if $pkgconfig('stb') {
+	#pkgconfig stb
+} $else {
+	#flag -I @VEXEROOT/thirdparty/stb_image
+}
+
 #include "stb_image.h"
 
 fn test_vroot_and_vmodroot() {
